draft-ietf-rats-corim.md

```diff
@@ -2070,133 +2042,195 @@ Logically, new ECT entries are appended to the existing ACS.
+A Verifier SHALL create a "matched entries" set, and SHALL populate it with all ACS entries which matched the condition ECT.
+
+If the matched entries array is not empty, then the condition ECT matches the ACS.
+
+If the matched entries array is empty, then the condition ECT does not match the ACS.
+
+### Comparing a condition ECT against a single ACS entry {#sec-match-one-condition-ect}
+
+A Verifier SHALL perform all of the comparisons in sections {{sec-compare-environment}), {{sec-compare-authority}), {{sec-compare-cm}} and {{sec-compare-element-list}).
+Each of these comparisons compares one field in the condition ECT against the same field in the ACS entry.
+
+If all three of the fields match then the condition ECT matches the ACS entry.
+
+If any of the fields does not match, then the condition ECT does not match the ACS entry.
+
+### Environment Comparison {#sec-compare-environment}
+
+A Verifier SHALL compare each field which is present in the condition ECT `environment-map` against the corresponding field in the ACS entry `environment-map` using binary comparison.
+Before performing the binary comparison, a Verifier SHOULD convert both `environment-map` fields into a form which meets CBOR Core Deterministic Encoding Requirements {{-cbor}}.
+
+If all fields which are present in the condition ECT `environment-map` are present in the ACS entry and are binary identical, then the environments match.
+
+If any field which is present in the condition ECT `environment-map` is not present in the ACS entry then the environments do not match.
+
+If any field which is present in the condition ECT `environment-map` is not binary identical to the corresponding ACS entry field then the environments do not match.
+
+If a field is not present in the condition ECT `environment-map` then the presence of, and value of, the corresponding ACS entry field does not affect whether the environments match.
+
+### Authority comparison {#sec-compare-authority}
+
+A Verifier SHALL compare the condition ECT expected `authority` value to the candidate entry `authority` value.
+
+If every entry in the condition ECT `authority` has a matching entry in the ACS entry `authority` field then the authorities match.
+The order of the fields in each `authority` field do not affect the result of the comparison.
+
+If any entry in the condition ECT `authority` does not have a matching entry in the ACS entry `authority` field then the authorities do not match.
+
+When comparing two `$crypto-key-type-choice` fields for equality, a Verifier SHALL treat them as equal if their deterministic CBOR encoding is binary equal.
+
+A Verifier MAY treat two keys as equal if they have different formats but represent the same key.
+For example, a Verifier may contain code to compare a key fingerprint against the key which, when hashed, creates that fingerprint.
```

Comment:
```suggestion

```
This kind of optional behavior could result in different verifiers arriving at a different result given the same inputs.
It is safer if the Verifier simply compared the "keys" (aka authority values) as opaque values rather than try to do format translation to a universal form.

Reply:
This decision could cause problems though. Since we’re requiring reference values to always include the expected root/intermediate/leaf expected signer from the attester (such as the sev-snp vcek, ask, or ark), then the reference value provider would want to avoid the bloat of copying the same full cert everywhere and just use a thumbprint. Is the expectation that the evidence translation will populate authorized-by with every format imaginable for describing the key in order to match? Or is the expectation for the profile to dictate a small number of supported formats that it will enumerate in the translation?

Reply:
I think we need a further change later to tidy up the processing of authority, I will remove this text now and we can add something like it later if that is the direction we want to follow.
I would like to see richer semantics available in the final draft.